presto-main/src/main/java/io/prestosql/execution/DynamicFilterConfig.java

```diff
@@ -0,0 +1,246 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.execution;
+
+import io.airlift.configuration.Config;
+import io.airlift.configuration.DefunctConfig;
+import io.airlift.configuration.LegacyConfig;
+import io.airlift.units.DataSize;
+import io.airlift.units.Duration;
+import io.airlift.units.MaxDataSize;
+import io.airlift.units.MaxDuration;
+import io.airlift.units.MinDuration;
+
+import javax.validation.constraints.NotNull;
+
+import static io.airlift.units.DataSize.Unit.KILOBYTE;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+@DefunctConfig({
+        "dynamic-filtering-max-per-driver-row-count",
+        "experimental.dynamic-filtering-max-per-driver-row-count",
+        "dynamic-filtering-max-per-driver-size",
+        "experimental.dynamic-filtering-max-per-driver-size",
+        "dynamic-filtering-range-row-limit-per-driver"
+})
+public class DynamicFilterConfig
+{
+    private boolean enableDynamicFiltering = true;
+    private boolean enableLargeDynamicFilters;
+    private Duration dynamicFilteringRefreshInterval = new Duration(200, MILLISECONDS);
+
+    private int smallBroadcastMaxPerDriverRowCount = 100;
+    private DataSize smallBroadcastMaxPerDriverSize = DataSize.of(10, KILOBYTE);
+    private int smallBroadcastRangeRowLimitPerDriver;
```

Comment:
let's use replicated instead of broadcast. That's how we call DFs in rest of codebase

Reply:
I used broadcast instead of replicated here because in FeatureConfig the user facing strings use broadcast (`join-distribution-type=BROADCAST` and `join-max-broadcast-table-size`)